### ISSUE TITLE
Intellisense not working on element from array

### ISSUE THREAD

**Issue by aka-STInG** [HaxeFoundation/intellij-haxe#602]
E.g.
```
var arr:Array<SomeType> = new Array<SomeType>();
var x = arr[i]
x.someMethod(); <- unresolved symbol
```

**Comment by EricBishton** [comment]
I think that your code sample does not show the error you are describing.  The x=arr[i] has nothing to do with the problem because you are accessing someMethod() on the array, not the element.  Is this what you intended to describe?

The best way to demonstrate a problem is by creating a working minimal example using try.haxe.org.

As it happens, I'm working on this class of problem right now, and it's quite tricky.  (Because the resolver code is NOT straightforward in the plugin.  I'd like to re-write it, but that's at least a two week project -- if not two months -- with a lot of edge cases, I fear.)

Just for the record, what platform, IDEA and plugin version are you using?

**Comment by aka-STInG** [comment]
Yes, sorry. I edited the sample.
Idea is 2016.3.6 and plugin is the latest one available to install from Idea.

**Comment by EricBishton** [comment]
If this is an accurate representation of the error, then this is fixed in 0.11.0-RC3:
```haxe
package ;
class TestIssue602 {
    public function new() {
    }
    public static function main() {
        var arr:Array<SomeType> = new Array<SomeType>();
        var x = arr[i]
        x.someMethod(); //<- unresolved symbol
    }
}

class SomeType {
    public function someMethod() {}
}
```
